src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -2,76 +2,129 @@
  * Copyright OpenSearch Contributors
  * SPDX-License-Identifier: Apache-2.0
  */
-import React, { useMemo, useState } from 'react';
+import React, { useMemo, useState, useEffect } from 'react';
 import * as Rx from 'rxjs';
+import moment from 'moment';
 import {
-  EuiPopover,
-  EuiTextColor,
+  EuiPanel,
   EuiListGroup,
   EuiListGroupItem,
   EuiTitle,
+  EuiPopoverTitle,
+  EuiIcon,
   EuiText,
+  EuiButtonEmpty,
   EuiSpacer,
   EuiHeaderSectionItemButtonProps,
   EuiButtonIcon,
+  EuiRadioGroup,
+  EuiTextColor,
+  EuiPopover,
   EuiToolTip,
 } from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import useObservable from 'react-use/lib/useObservable';
-import { ChromeRecentlyAccessedHistoryItem } from '../..';
+import { SavedObjectsNamespaceType } from 'src/core/public';
+import { ChromeRecentlyAccessedHistoryItem, SavedObject } from 'opensearch-dashboards/public';
 import { WorkspaceObject } from '../../../workspace';
-import { createRecentNavLink } from './nav_link';
 import { HttpStart } from '../../../http';
-import { ChromeNavLink } from '../../../';
 import './recent_items.scss';
 
+const widthForRightMargin = 8;
+
 export interface Props {
   recentlyAccessed$: Rx.Observable<ChromeRecentlyAccessedHistoryItem[]>;
   workspaceList$: Rx.Observable<WorkspaceObject[]>;
   navigateToUrl: (url: string) => Promise<void>;
-  basePath: HttpStart['basePath'];
-  navLinks$: Rx.Observable<ChromeNavLink[]>;
   renderBreadcrumbs: React.JSX.Element;
   buttonSize?: EuiHeaderSectionItemButtonProps['size'];
+  http: HttpStart;
 }
 
+interface SavedObjectMetadata {
+  icon?: string;
+  title?: string;
+  editUrl?: string;
+  inAppUrl?: { path: string; uiCapabilitiesPath: string };
+  namespaceType?: SavedObjectsNamespaceType;
+}
+export type SavedObjectWithMetadata<T = unknown> = SavedObject<T> & {
+  meta: SavedObjectMetadata;
+};
+
+type DetailedRecentlyAccessedItem = ChromeRecentlyAccessedHistoryItem &
+  SavedObjectWithMetadata &
+  ChromeRecentlyAccessedHistoryItem['meta'] & {
+    updatedAt: number;
+    workspaceName?: string;
+  };
+
+export const bulkGetDetail = (
```

Comment:
What about move this into a utils file and rename to some others like bulkGetSavedObjectsDetail, which could better indicate the effect of this function. And we don't need to export if nowhere imports this.

Reply:
Thanks for the review, I delete export and I think it's better to keep it here as the required type SavedObjectWithMetadata has the import restriction. Here I just simply duplicate it in the file. [ Unexpected path "src/plugins/saved_objects_management/common/types" imported in restricted zone.]